test/irmin/test.ml

```diff
@@ -103,6 +103,9 @@ let test_bin () =
     (T.size_of l [ "foo"; "bar" ]);
   let s = T.of_bin_string l "foobar" in
   Alcotest.(check (ok tl)) "decode list" (Ok [ "foo"; "bar" ]) s;
+  Alcotest.check_raises "decode error"
+    (Assert_failure ("src/irmin/type.ml", 1555, 2))
+    (fun () -> ignore (T.of_bin_string l "foobarbar"));
```

Comment:
This is a nasty API to be exposing to users, but this can be fixed in another PR.

Reply:
I guess this is related to #483